h/api/uri.py

```diff
@@ -64,6 +64,7 @@
 """
 import re
 
+from h._compat import text_type
```

Comment:
:+1:
Duplicated the relevant bits (also looked for other `h._compat` occurrences in `h.api`) to `h.api._compat` and changed the import paths.